webapps-backup/src/main/java/io/camunda/webapps/backup/repository/opensearch/OpensearchBackupRepository.java

```diff
@@ -0,0 +1,475 @@
+import static io.camunda.webapps.backup.repository.opensearch.OpensearchRequestDSL.repositoryRequestBuilder;
+import static io.camunda.webapps.backup.repository.opensearch.SnapshotState.FAILED;
+import static io.camunda.webapps.backup.repository.opensearch.SnapshotState.PARTIAL;
+import static io.camunda.webapps.backup.repository.opensearch.SnapshotState.STARTED;
+import static io.camunda.webapps.backup.repository.opensearch.SnapshotState.SUCCESS;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+
+import io.camunda.webapps.backup.BackupRepository;
+import io.camunda.webapps.backup.BackupService;
+import io.camunda.webapps.backup.BackupStateDto;
+import io.camunda.webapps.backup.GetBackupStateResponseDetailDto;
+import io.camunda.webapps.backup.GetBackupStateResponseDto;
+import io.camunda.webapps.backup.Metadata;
+import io.camunda.webapps.backup.exceptions.InvalidRequestException;
+import io.camunda.webapps.backup.exceptions.ResourceNotFoundException;
+import io.camunda.webapps.backup.repository.BackupRepositoryProps;
+import io.camunda.webapps.backup.repository.GenericBackupException;
+import io.camunda.webapps.backup.repository.SnapshotNameProvider;
+import io.camunda.webapps.util.ExceptionSupplier;
+import java.net.SocketTimeoutException;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.opensearch.client.json.JsonData;
+import org.opensearch.client.opensearch.OpenSearchAsyncClient;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.opensearch._types.OpenSearchException;
+import org.opensearch.client.opensearch.snapshot.GetSnapshotResponse;
+import org.opensearch.client.opensearch.snapshot.SnapshotInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OpensearchBackupRepository implements BackupRepository {
+  public static final String SNAPSHOT_MISSING_EXCEPTION_TYPE = "snapshot_missing_exception";
+  public static final String REPOSITORY_MISSING_EXCEPTION_TYPE = "repository_missing_exception";
+  private static final Logger LOGGER = LoggerFactory.getLogger(OpensearchBackupRepository.class);
+
+  private final OpenSearchClient openSearchClient;
+  private final OpenSearchAsyncClient openSearchAsyncClient;
+
+  private final BackupRepositoryProps backupProps;
+  private final SnapshotNameProvider snapshotNameProvider;
+
+  public OpensearchBackupRepository(
+      final OpenSearchClient openSearchClient,
+      final OpenSearchAsyncClient openSearchAsyncClient,
+      final BackupRepositoryProps operateProperties,
```

Comment:
No `ObjectMapper`  injected from the context, for that we use the `JsonpMapper` from `client._transport.jsonpMapper()` (even if potentially it's not needed?)